pyhanko/cli.py

```diff
@@ -517,6 +523,8 @@ def validate_signatures(ctx, infile, executive_summary,
             else:
                 print('%s:%s:%s' % (name, fingerprint, status_str))
 
+            exit_if_invalid(signature_ok)
```

Comment:
Same comment as before regarding the `exit_if_invalid` call.

But I also don't think it's appropriate to abort validation on purpose partway through a `for` loop. I'd prefer collecting the `signature_ok` values for each field, and then exiting with an exit code after we're done validating all (regular) signatures.

Reply:
I've switched from `exit_if_invalid` to `raise click.ClickException`, and letting the loop finish before exiting.

Do you think the message `raise click.ClickException("Validation failed")` is fine, so the error message shown to the user is `Error: Validation failed`?